src/server/snap.rs

```diff
@@ -171,6 +174,28 @@ fn send_snap(
     res
 }
 
+/// Write binary snapshot to local disk, use RateLimiter to control the speed
+fn write_snapshot<W: Write>(w: &mut W, limiter: Arc<RateLimiter>, data: &[u8]) -> io::Result<()> {
```

Comment:
IMO, I prefer supporting a RateLimitWriter which implements io.Write for later use.

Reply:
Will do.